WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+SectionHelpers.swift

```diff
@@ -157,4 +163,65 @@ extension BlogDetailsViewController {
         let listViewController = PluginListViewController(site: site, query: query)
         presentationDelegate?.presentBlogDetailsViewController(listViewController)
     }
- }
+}
+
+struct ApplicationPasswordRequiredView<Content: View>: View {
+    private let blog: Blog
+    private let localizedFeatureName: String
+    @State private var site: WordPressSite?
+    private let builder: (WordPressClient) -> Content
+
+    init(blog: Blog, localizedFeatureName: String, @ViewBuilder content: @escaping (WordPressClient) -> Content) {
+        wpAssert(blog.account == nil, "The Blog argument should be a self-hosted site")
+
+        self.blog = blog
+        self.localizedFeatureName = localizedFeatureName
+        self.site = try? WordPressSite(blog: blog)
+        self.builder = content
+    }
+
+    var body: some View {
+        if let site {
+            builder(WordPressClient(site: site))
+        } else {
+            RestApiUpgradePrompt(localizedFeatureName: localizedFeatureName) {
+                Task {
+                    await self.migrate()
+                }
+            }
+        }
+    }
+
+    @MainActor
+    private func migrate() async {
+        guard let url = try? blog.getUrlString() else {
+            let error = NSLocalizedString("applicationPasswordMigration.error.siteUrlNotFound", value: "Cannot find the current site's url", comment: "Error message when the current site's url cannot be found")
```

Comment:
(nit) Move to the bottom to `Strings` (applies to all other instances too)

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23726/commits/53aaea9357682fec8f159467a3b57c7c5718d40c